qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/RolesAndPrivilegesSteps.java

```diff
@@ -0,0 +1,161 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ * 
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.contrib.qaframework.automation;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import static org.junit.Assert.assertTrue;
+
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.contrib.qaframework.page.AddEditNewPrivilegePage;
+import org.openmrs.contrib.qaframework.page.AddNewRolePage;
+import org.openmrs.contrib.qaframework.page.AdministrationManageRolesPage;
+import org.openmrs.contrib.qaframework.page.AdministrationPage;
+import org.openmrs.contrib.qaframework.page.ConfigureMetadataPage;
+import org.openmrs.contrib.qaframework.page.ManagePrivilegesPage;
+import org.openmrs.contrib.qaframework.page.SystemAdministrationPage;
+
+public class RolesAndPrivilegesSteps extends Steps {
+
+	private static final String PRIVILEGE_NAME = "PrivilegeTest";
+	private static final String ROLE_NAME = "appletest";
```

Comment:
is it possible to save a privilege or role with a name that exists in the system?

Reply:
yes and no! i say yes because you can add the exact name of the existing role/privilege and i say no because what the system does is instead of creating two roles/privileges with the same name, what it does is get the existing role/privilege and edit it. so when you save it will just update the existing role/privilege (**with a similar name as the one you are trying to create**)

Reply:
> so when you save it will just update the existing role/privilege (**with a similar name as the one you are trying to create**)

This would be a good ideal for an automated test. I think using random string value for ROLE_NAME, and PRIVILEGE_NAME would be a good approach to ensure on each run a unique string values are recorded in the system. This would mean even during the editing these values, a randomly generated strings should be passed as new values so as to avoid having roles and privileges in the system that have similar names.

Reply:
> to avoid having roles and privileges in the server that have similar names

this cannot happen but having randomly generated names is ideal. i will surely update the work flow with it. thanks